flocker/control/_persistence.py

```diff
@@ -279,10 +279,13 @@ def update_leases(transform, persistence_service):
         persistence service has saved.
     """
     config = persistence_service.get()
-    new_config = config.set("leases", transform(config.leases))
-    d = persistence_service.save(new_config)
-    d.addCallback(lambda _: new_config.leases)
-    return d
+    new_leases = transform(config.leases)
+    if config.leases != new_leases:
```

Comment:
The safe way to get this optimization would probably be to have pyrsistent implement equality by starting with an identity check.  This puts the complexity where it is minimized and provides a wider benefit at the same time.